File: TenMinsPy/day104.py
# sortColumn(rows, columnName) sorts rows array according to column specified, ascending alphabethical
import csv

def sort_column(column_name):
    file_path = "D:/coding30minsAday/textFiles/JTrialCsv.csv"
    
    try:
        # Read all content from the file
        with open(file_path, 'r', newline='', encoding='utf-8') as f:
            reader = csv.reader(f)
            lines = list(reader)
        
        # Check if file is empty
        if not lines:
            return

        # Extract header and rows
        header = lines[0]
        rows = lines[1:]
        
        # Find the column index
        try:
            index = header.index(column_name)
        except ValueError:
            print(f"Column '{column_name}' not found.")
            return

        # Sort the rows based on the column index (case-insensitive)
        # Lambda mimics: rowA[index].compareToIgnoreCase(rowB[index])
        rows.sort(key=lambda x: x[index].lower())

        # Combine header and sorted rows
        lines = [header] + rows

        # Write the sorted content back to the file
        with open(file_path, 'w', newline='', encoding='utf-8') as f:
            writer = csv.writer(f)
            writer.writerows(lines)
            
    except Exception as e:
        print(f"Error: {e}")

File: TenMinsJava/day104.java
// sortColumn(rows, columnName) sorts rows array according to column specified, ascending alphabethical
package TenMinsJava;

import java.io.File;
import java.io.PrintWriter;
import java.util.ArrayList;
import java.util.Scanner;

public class day104 {
    public static void sortColumn(String columnName) {
        try {
            String filePath = "D:/coding30minsAday/textFiles/JTrialCsv.csv";
            File f = new File(filePath);
            ArrayList<String> lines = new ArrayList<>();
            Scanner fScan =  new Scanner(f);
            while (fScan.hasNextLine()) {
                lines.add(fScan.nextLine());
            }
            String head = lines.get(0);
            String header[] = lines.get(0).split(",");
            // find the column index
            int tempCouter = -1;
            for (int i = 0; i < header.length; i++) {
                if (header[i].equals(columnName)) {
                    tempCouter = i;
                    break;
                }
            }
            int index = tempCouter;
            lines.remove(0);
            lines.sort((a, b) -> {
                String[] rowA = a.split(",");
                String[] rowB = b.split(",");
                return rowA[index].compareToIgnoreCase(rowB[index]);
            });
            lines.add(0, head);
            
            PrintWriter newF = new PrintWriter(f);
            for (String r : lines) {
                newF.println(r);
            }
            newF.close();
        } catch (Exception e) {
            System.err.println(e);
        }
    }
}